components/core/src/clp_s/SchemaTree.hpp

```diff
@@ -106,8 +112,23 @@ class SchemaTree {
         m_node_map.clear();
     }
 
+    /**
+     * Finds the root node for a subtree matching a given type given the root node for some subtree
+     * in which the subtree we are looking for can be found, and some descendent node of the subtree
```

Comment:
That's fair. Open to any ideas for describing this more clearly.

Generally what this is trying to do is perform a special kind of search within some subtree. Specifically between the root of some subtree and a descendent somewhere in that subtree we want to find the node along the path between that descendent and the root that matches a given node type, and if there are multiple such nodes we want to return the one closest to the root of the subtree.

The reason we want this is a bit easier to understand in the context of the generic parser feature. For that feature I was finding that for marshalling a record with complicated nested structure I really needed to know a) the root of the subtree in the MPT for the generic parser b) the schema of the generic parser and c) the column readers for the parser in order. This function solves (a) in a way that makes it really easy to provide the same semantic to some other generic parser in its subtree since it knows its own root and can easily find some descendent column of another generic parser by scanning the schema belonging to that generic parser (which is a subset of its own schema).